allows matlab to be an optional.

RussTedrake opened the pull request:
now mex_setup() must be called from the referencing CMakeLists.txt

steinachim commented:
I just tried setting up a new installation and this pull request seems to break the installation instructions for "staying up to date" in the wiki: 
When running just
```
cd drake
git pull origin master
cd ..
make
```
the build fails with the following error:
```
-------------------------------------------
-- drake
-------------------------------------------

BUILD_PREFIX: /home/steinachim/MyDrakeTest/build


-- The C compiler identification is GNU 4.8.2
-- The CXX compiler identification is GNU 4.8.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found Java: /usr/bin/java (found version "1.7.0.75") 
CMake Warning at cmake/mex.cmake:227 (message):
  Your MEX compiler flags contained '-ansi', but we've removed that flag for
  compatibility with C++11
Call Stack (most recent call first):
  cmake/mex.cmake:361 (mex_setup)
  CMakeLists.txt:56 (include)


CMake Warning at cmake/mex.cmake:227 (message):
  Your MEX compiler flags contained '-ansi', but we've removed that flag for
  compatibility with C++11
Call Stack (most recent call first):
  CMakeLists.txt:57 (mex_setup)


CMake Error at cmake/mex.cmake:239 (add_library):
  add_library cannot create target "liblast" because another target with the
  same name already exists.  The existing target is a static library created
  in source directory "/home/steinachim/MyDrakeTest/drake".  See
  documentation for policy CMP0002 for more details.
Call Stack (most recent call first):
  CMakeLists.txt:57 (mex_setup)
```

Calling `git pull origin master` in both the `spotless/cmake` and `drake/cmake` directory fixes the issue. Since I'm not sure about the internal build processes of Drake, I can't say if this is a symptom of a larger issue or if simply an update of the installation manual is necessary.

RussTedrake commented:
I'll update the wiki.

You need to `git pull origin master --recurse-submodules` and `git submodule update --init`


On Apr 16, 2015, at 6:50 AM, Achim Stein <notifications@github.com> wrote:

> I just tried setting up a new installation and this pull request seems to break the installation instructions for "staying up to date" in the wiki: 
> When running just
> 
> cd drake
> git pull origin master
> cd ..
> make
> the build fails with the following error:
> 
> -------------------------------------------
> -- drake
> -------------------------------------------
> 
> BUILD_PREFIX: /home/steinachim/MyDrakeTest/build
> 
> 
> -- The C compiler identification is GNU 4.8.2
> -- The CXX compiler identification is GNU 4.8.2
> -- Check for working C compiler: /usr/bin/cc
> -- Check for working C compiler: /usr/bin/cc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Found Java: /usr/bin/java (found version "1.7.0.75") 
> CMake Warning at cmake/mex.cmake:227 (message):
>   Your MEX compiler flags contained '-ansi', but we've removed that flag for
>   compatibility with C++11
> Call Stack (most recent call first):
>   cmake/mex.cmake:361 (mex_setup)
>   CMakeLists.txt:56 (include)
> 
> 
> CMake Warning at cmake/mex.cmake:227 (message):
>   Your MEX compiler flags contained '-ansi', but we've removed that flag for
>   compatibility with C++11
> Call Stack (most recent call first):
>   CMakeLists.txt:57 (mex_setup)
> 
> 
> CMake Error at cmake/mex.cmake:239 (add_library):
>   add_library cannot create target "liblast" because another target with the
>   same name already exists.  The existing target is a static library created
>   in source directory "/home/steinachim/MyDrakeTest/drake".  See
>   documentation for policy CMP0002 for more details.
> Call Stack (most recent call first):
>   CMakeLists.txt:57 (mex_setup)
> Calling git pull origin master in both the spotless/cmake and drake/cmake directory fixes the issue. Since I'm not sure about the internal build processes of Drake, I can't say if this is a symptom of a larger issue or if simply an update of the installation manual is necessary.
> 
> —
> Reply to this email directly or view it on GitHub.
>

steinachim commented:
Thanks for the fast reply! Everything seems to be working now :)
